### PR TITLE
Support function array/reshape

### PR DESCRIPTION
**Pull request by youchan**
Support function array/reshape

## example

```ruby
> require 'chainer'
> x = Numo::SFloat.new(4,3, 2).seq
> h = Chainer::Functions::Array::Reshape.reshape(x, [2,2,6])
> h.data
=> Numo::SFloat#shape=[2,2,6]
[[[0, 1, 2, 3, 4, 5],
  [6, 7, 8, 9, 10, 11]],
 [[12, 13, 14, 15, 16, 17],
  [18, 19, 20, 21, 22, 23]]]
```

This commit depends on #60

### CODE REVIEW

File: lib/chainer/functions/array/reshape.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      class Reshape < Function
+        def initialize(shape)
+            @shape = shape
+        end
+
+        def self.reshape(xs, shape)
+          self.new(shape).(xs)
+        end
+
+        def forward_cpu(xs)
```
Comment (naitoh) [inline lib/chainer/functions/array/reshape.rb]: Here, you define the forward_cpu(), backward_cpu() method.
If there is no plans to separate the code between the CPU and the GPU in the future, I think that the forward(), backward() names are better as in the original chainer/functions/array/reshape.py.

**Comment by hatappi** [comment]
Please resolve conflicts 🙏

**Comment by hatappi** [comment]
It looks good to me 👍 

@naitoh 
What do you think about this PR?

**Comment by naitoh** [comment]
@hatappi It looks good to me, too. 👍

**Comment by hatappi** [comment]
@naitoh 
Thank you for confirmation.

**Comment by hatappi** [comment]
@youchan 
Thank you for contributing!!!

**Pull request merged**